refreshVersions/src/main/kotlin/de/fayard/versions/internal/VersionsPlaceholdersReplacement.kt

```diff
@@ -30,12 +32,23 @@ internal fun Project.setupVersionPlaceholdersResolving() {
     val versionKeyReader = retrieveVersionKeyReader()
     var properties: Map<String, String> = project.getVersionProperties()
     allprojects {
+        val project : Project = this
+
         configurations.all {
+            val configuration: Configuration = this
             @Suppress("UnstableApiUsage")
             withDependencies {
-                val dependenciesToReplace = filter { it is ModuleDependency && it.version == versionPlaceholder }
+                val dependencies = filterIsInstance<ModuleDependency>()
+
+                val dependenciesToReplace = dependencies.filter { it.version == versionPlaceholder }
                 removeAll(dependenciesToReplace)
-                for (dependency in dependenciesToReplace) {
+
+                if (dependencies.any { it.version != versionPlaceholder }) {
+                    val warnFor = (dependencies - dependenciesToReplace).take(3).map { it.name }
+                    logger.warn(""":${project.name}:${configuration.name} found hardcoded dependencies $warnFor   See https://github.com/jmfayard/refreshVersions/issues/160 """)
+                }
+
+                for (dependency in dependencies) {
```

Comment:
This currently messes up the versions.properties file and brings two sources of truth because the currently used version is not retrieved.

Let's change that in another PR, all while adding unit tests along the way.